auth/design.md

```diff
@@ -0,0 +1,68 @@
+﻿
+# etcd v3 authentication design
+
+## Why not using the existing v2 auth mechanism?
+The v3 protocol uses gRPC as its transport instead of RESTful interface like v2. This new protocol provides an opportunity to iterate on and improve the v2 design. For example, v3 auth has connection based authentication, rather than v2's slower per-request authentication. Additionally, v2 auth's semantics tend to be unwieldy in practice with respect to reasoning about consistency, which will be described in the next sections. For v3, there is a well-defined description and implementation of the authentication mechanism which fixes the deficiencies in the v2 auth system.
+
+### Functionality requirements
+1. per connection authentication, not per request
+   1. User ID + password based authentication implemented on gRPC API
+   1. authentication must be refreshed after auth information update
+1. its functionality should be as simple and useful as v2
+   1. v3 provides a flat key space, unlike the directory structure of v2. Permission check will be provided as interval matching.
+1. it should avoid the consistency problem of v2, described later
+
+### Main required changes
+1. a client needs to create a dedicated connection just for authentication before sending actual requests

```

Comment:
s/1. a/1. A
s/needs to/must
s/just/only
s/actual/authenticated

These can be `*` instead of `1.` since there's no ordering?